primitives/runtime/src/offchain/storage.rs

```diff
@@ -28,6 +28,25 @@ pub struct StorageValueRef<'a> {
 	kind: StorageKind,
 }
 
+/// Reason for not being able to provide the stored value
+#[derive(Debug, PartialEq, Eq)]
+pub enum StorageRetrivalError {
+	/// Key not found in local storage
+	NoKey,
+	/// Value found but undecodable
+	Undecodable,
+}
+
+/// Possible errors when mutating a storage value.
+#[derive(Debug, PartialEq, Eq)]
+pub enum MutateStorageError<T, E> {
+	/// The new value might not encode correctly or the underlying db failed to update.
```

Comment:
Neither encode nor storing can fail.

What fails is setting the value in the compare and set instruction.

Reply:
Great point. Will rename one branch to CompareAndSetFailed